examples/compiled/bar_diverging_stack_population_pyramid.vg.json

```diff
@@ -49,19 +49,63 @@
   ],
   "marks": [
     {
-      "name": "marks",
-      "type": "rect",
-      "style": ["bar"],
-      "from": {"data": "data_0"},
+      "type": "group",
+      "from": {
+        "facet": {
+          "data": "data_0",
+          "name": "stack_group_main",
+          "groupby": ["age"],
+          "aggregate": {
+            "fields": [
+              "sum_signed_people_start",
+              "sum_signed_people_start",
+              "sum_signed_people_end",
+              "sum_signed_people_end"
+            ],
+            "ops": ["min", "max", "min", "max"]
+          }
+        }
+      },
       "encode": {
         "update": {
-          "fill": {"scale": "color", "field": "gender"},
-          "x": {"scale": "x", "field": "sum_signed_people_end"},
-          "x2": {"scale": "x", "field": "sum_signed_people_start"},
           "y": {"scale": "y", "field": "age"},
-          "height": {"scale": "y", "band": true}
+          "height": {"scale": "y", "band": true},
+          "x": {
+            "signal": "min(scale('x',datum[\"min_sum_signed_people_start\"]),scale('x',datum[\"max_sum_signed_people_start\"]),scale('x',datum[\"min_sum_signed_people_end\"]),scale('x',datum[\"max_sum_signed_people_end\"]))"
+          },
+          "x2": {
+            "signal": "max(scale('x',datum[\"min_sum_signed_people_start\"]),scale('x',datum[\"max_sum_signed_people_start\"]),scale('x',datum[\"min_sum_signed_people_end\"]),scale('x',datum[\"max_sum_signed_people_end\"]))"
+          },
+          "clip": {"value": true}
```

Comment:
We just want to support clipping for corner radius of stacked bars, but somehow we modify all of bar specification to include aggregate and clipping.  I think that's a bit of an overkill.  

I'd rather generate these complicated stuff only when they're needed. Otherwise, this dilute our story for moving from VL to VG since our generated VG are unnecessarily complicated.

(In general, we should only generate Vega diff / SVG only when necessary.)

Reply:
We are only doing that for stacked bars --- this one uses stack for the x axis.

Reply:
But if there is no corner radius, we don't really need to do clipping? 

Besides spec complexity, it probably makes it a bit slower to render?
I'm more concerned about spec complexity though.  It's kinda like we're handling corner case here, but by doing that we shouldn't make the common case worse. :)

Reply:
I see, good point!